cli/stage_parser.py

```diff
@@ -49,9 +94,18 @@ def render(file_name):
         """
         err = StringIO.StringIO()
         out = StringIO.StringIO()
+        exception = None
         with redirect_stderr(err):
             with redirect_stdout(out):
-                result = __caller__.cmd('slsutil.renderer', file_name)
+                try:
+                    result = __caller__.cmd('slsutil.renderer', file_name)
+                except salt.exceptions.SaltException as ex:
+                    exception = StageRenderingException(file_name, ex.strerror)
```

Comment:
Why not raise the exception here? Or does that cause trouble becuase stderr and stdout are redirected?

Reply:
Right, if I raise the exception here, all the output made by the code that catches the exception is redirected.